dev/tasks/crossbow.py

```diff
@@ -706,7 +706,25 @@ def get_version(root, **kwargs):
         'git describe --dirty --tags --long --match "apache-arrow-[0-9].*"'
     )
     version = parse_git_version(root, **kwargs)
-    return version.format_next_version(guess_next_version)
+    next_version = version.format_next_version(guess_next_version)
+    # Ensure using ${MAJOR}.${MINOR}.${PATCH}-${SUFFIX} for pre-release version
+    # number. Because NuGet package requires this format for version number:
+    # https://docs.microsoft.com/en-us/nuget/concepts/package-versioning
+    #
+    # NuGet package version number follows Semantic Versioning 1.0.0:
+    # https://semver.org/spec/v1.0.0.html
+    #
+    # > A pre-release version number MAY be denoted by appending an
+    # > arbitrary string immediately following the patch version and a
+    # > dash. The string MUST be comprised of only alphanumerics plus
+    # > dash [0-9A-Za-z-].
+    #
+    # Example:
+    #
+    #   '0.16.1.dev10' ->
+    #   '0.16.1-dev10'
+    next_version = re.sub(r'\.(dev\d+)$', r'-\1', next_version)
+    return next_version
```

Comment:
@kszucs What do you think about this? Is it acceptable?

Reply:
Looks fine for me, but we need to test whether the wheels and conda pkgs accept this format.

Reply:
OK. I'll run jobs for them.

Reply:
Umm... Conda doesn't like it:

```text
Error: bad character '-' in package/version: 0.17.1-dev124
```

Reply:
I've added `arrow.no_rc_semver_version` instead of changing the current `arrow.version`.